### PR TITLE
Fix Jetpack preview authentication

### LINKED ISSUE #3637: Previews on Jetpack sites show a login screen
I think this is a side effect of #3597, since now Jetpack sites will have an `authToken` and `WPURLRequest` will remove the password if there's a token

Quick patch: https://gist.github.com/14a56fad8c5990b44ce5

### PR DESCRIPTION
**Pull request by koke**
As a side effect of #3597, `blog.authToken` was not `nil` for Jetpack
sites. When there's an auth token, WPURLRequest will drop the password
from the request. Since the login was posted to the self hosted and not
WordPress.com, the authentication failed.

This sets the token only if the blog is a WordPress.com one, and adds a
safeguard to prevent tokens from being sent to non-wp.com URLs.

fixes #3637 

/cc @jleandroperez since I think you wrote WPURLRequest
/cc @sendhil since I mentioned it in private

### CODE REVIEW

File: WordPress/Classes/Utility/WPURLRequest.m [hunk 1]
```diff
@@ -22,6 +22,12 @@ + (NSURLRequest *)requestForAuthenticationWithURL:(NSURL *)loginUrl
     NSParameterAssert(password != nil || bearerToken != nil);
     
     NSMutableURLRequest *request = [self mutableRequestWithURL:loginUrl userAgent:userAgent];
+
+    NSString *hostname = [loginUrl host];
+    if (![hostname isEqualToString:@"wordpress.com"] && ![hostname hasSuffix:@".wordpress.com"]) {
```
Comment (jleandroperez) [inline WordPress/Classes/Utility/WPURLRequest.m]: You can use [this helper](https://github.com/wordpress-mobile/WordPress-iOS/blob/develop/WordPress/Classes/Categories/NSString%2BHelpers.m#L191) instead

**Comment by jleandroperez** [comment]
@koke (unless i'm missing something) the **WPURLRequest** helper might break, when dealing with *WordPress.com* sites, with a custom domain.

**Comment by sendhil** [comment]
@koke - I branched off `release/5.2` earlier, but once this PR is sorted out could you re-submit it with the changes to that branch so it can make it into the next release? As we merge the release branch into develop every few days it'll be cleaner I think than cherry picking from `develop`.

**Comment by koke** [comment]
@jleandroperez `WPURLRequest` should be using the non-mapped domain. Otherwise, it can't use HTTPS. If it's not using HTTPS, it shouldn't be sending wp.com tokens *or* passwords

File: WordPress/Classes/Utility/WPURLRequest.m (hunk 1 above)
Comment (koke) [inline WordPress/Classes/Utility/WPURLRequest.m]: That helper takes a string, and I already have a `NSURL` that doesn't need parsing

**Comment by jleandroperez** [comment]
Looks good @koke, nice catch!

**Pull request merged**